src/ring_theory/coprime.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Ken Lee, Chris Hughes
+-/
+
+import algebra.big_operators
+import data.fintype.basic
+import data.set.disjointed
+
+/-!
+# Coprime elements of a ring
+
+## Main definitions
+
+* `is_coprime x y`: that `x` and `y` are coprime, defined to be the existence of `a` and `b` such
+that `a * x + b * y = 1`.
+
+-/
+
+open_locale classical big_operators
+
+universes u v
+
+variables {R : Type u} [comm_semiring R] (x y z : R)
+
+/-- The proposition that `x` and `y` are coprime, defined to be the existence of `a` and `b` such
+that `a * x + b * y = 1`. -/
+@[simp] def is_coprime : Prop :=
+∃ a b, a * x + b * y = 1
+
+variables {x y z}
+
```

Comment:
Pleae add the following theorems (or a TODO item in the module docstring):
```lean
theorem is_coprime_iff_gcd {R} [euclidean_domain R] {x y : R} : is_coprime x y ↔ is_unit (gcd x y) := sorry
theorem nat.is_coprime_iff_coprime {m n : ℕ} : is_coprime m n ↔ nat.coprime m n := sorry
```
Another source of useful theorems is the list of theorems about `nat.coprime` in `data/nat/gcd`.

Reply:
the first one is in `ring_theory/euclidean_domain.lean`

Reply:
the second is false (we don't have subtraction in nat)